lib/common/model.js

```diff
@@ -257,25 +277,20 @@ function ModelFactory (
                     update.$set = {};
                 }
                 this.addWaterlineUpdatedAtToObject(update, '$set');
-                return self.runNativeMongo('findAndModify', _arguments);
-            }
-
-            if (update.$pullAll) {
-                if (!update.$set) {
-                    update.$set = {};
-                }
-                this.addWaterlineUpdatedAtToObject(update, '$set');
-                return self.runNativeMongo('findAndModify', _arguments);
+                return self.runNativeMongo('findAndModify', _arguments).then(function(doc){
+                    return doc ? self.deSerialize(doc) : doc;
+                });
             }
 
             this.addWaterlineUpdatedAtToObject(update);
             this.addWaterlineCreatedAtToObject(update);
 
             return Promise.fromNode(self.validate.bind(this, update))
             .then(function() {
-                return self.runNativeMongo('findAndModify', _arguments);
+                return self.runNativeMongo('findAndModify', _arguments).then(function(doc){
+                    return doc ? self.deSerialize(doc) : doc;
+                });
```

Comment:
Could it be possible to change doc null validation to deSerialize()? that would save the efforts every time when calling it.